### PR TITLE
feat: blacklist ignored skills

### PR DESCRIPTION
**Pull request by navinkarkera**
This commit adds settings to set threshold for ignored tags. The skills which cross these thresholds will be marked as blacklisted so that they are not presented to users for further verification.

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/management/commands/finalize_xblockskill_tags.py [hunk 1]
```diff
@@ -34,49 +34,93 @@ def add_arguments(self, parser):
...
         )
         parser.add_argument(
-            '--ratio-threshold',
-            help=_('Ratio of min votes to total votes for verification'),
+            '--ratio-verified-threshold',
+            help=_('Ratio of min verified_count to total count for verification'),
+        )
+        parser.add_argument(
+            '--min-ignored-votes',
+            help=_('Minimum number of times skill is not verified to be blacklisted'),
```
Comment (pkulkark) [inline taxonomy/management/commands/finalize_xblockskill_tags.py]:
```suggestion
            help=_('Minimum number of times the skill must be ignored to be blacklisted'),
```

File: taxonomy/management/commands/finalize_xblockskill_tags.py [hunk 2]
```diff
@@ -34,49 +34,93 @@ def add_arguments(self, parser):
...
+        )
+        parser.add_argument(
+            '--ratio-ignored-threshold',
+            help=_('Ratio of min ignored count to total count for blacklisting'),
```
Comment (pkulkark) [inline taxonomy/management/commands/finalize_xblockskill_tags.py]:
```suggestion
            help=_('Ratio of min ignored_count to total count for blacklisting'),
```

File: taxonomy/management/commands/finalize_xblockskill_tags.py [hunk 3]
```diff
@@ -34,49 +34,93 @@ def add_arguments(self, parser):
...
+    @staticmethod
+    def _get_and_validate_argument_value(argument, setting_key, options):
+        """
+        Gets argument value from arguments or settings. Raises error if None.
+        """
+        value = options.get(argument, None) or getattr(settings, setting_key, None)
+        if value is None:
+            raise InvalidCommandOptionsError(
+                f'Either configure {setting_key} in settings or pass with arg --{argument.replace("_", "-")}'
+            )
```
Comment (pkulkark) [inline taxonomy/management/commands/finalize_xblockskill_tags.py:62]: Would this exit the task after raising the error?

File: tests/management/test_finalize_xblockskill_tags.py [hunk 4]
```diff
@@ -46,7 +47,6 @@ def test_finalize_xblockskill_tags_without_unverified_skills(self):
                 ]
```
Comment (pkulkark) [inline tests/management/test_finalize_xblockskill_tags.py:47]: Please change the `call_command` here to not use the earlier options.

File: tests/management/test_finalize_xblockskill_tags.py [hunk 5]
```diff
@@ -77,7 +77,6 @@ def test_finalize_xblockskill_tags_below_minimum_votes(self):
...
```
Comment (pkulkark) [inline tests/management/test_finalize_xblockskill_tags.py:77]: Same here. The `call_command` here is still using the earlier options.

File: tests/management/test_finalize_xblockskill_tags.py [hunk 6]
```diff
@@ -111,11 +110,9 @@ def test_finalize_xblockskill_tags_below_ratio_threshold(self):
...
```
Comment (pkulkark) [inline tests/management/test_finalize_xblockskill_tags.py:110]: Here as well

File: tests/management/test_finalize_xblockskill_tags.py [hunk 7]
```diff
@@ -141,94 +138,70 @@ def test_finalize_xblockskill_tags(self):
                     'Xblockskill tags verification task is completed'
```
Comment (pkulkark) [inline tests/management/test_finalize_xblockskill_tags.py:138]: Same thing here.

File: taxonomy/management/commands/finalize_xblockskill_tags.py (hunk 3 above)
Comment (navinkarkera) [reply taxonomy/management/commands/finalize_xblockskill_tags.py:62]: Yes, raising an exception anywhere will exit the task if not caught. We also have a test for it [here](https://github.com/open-craft/taxonomy-connector/pull/5/files#diff-34a0582d52b032b514b6b6019bda6e5fee67efef7811f2fea388634b9490e123R145-R151).

File: tests/management/test_finalize_xblockskill_tags.py (hunk 4 above)
Comment (navinkarkera) [reply tests/management/test_finalize_xblockskill_tags.py:47]: Updated. Thanks!

File: tests/management/test_finalize_xblockskill_tags.py (hunk 5 above)
Comment (navinkarkera) [reply tests/management/test_finalize_xblockskill_tags.py:77]: Updated. Thanks!

File: tests/management/test_finalize_xblockskill_tags.py (hunk 6 above)
Comment (navinkarkera) [reply tests/management/test_finalize_xblockskill_tags.py:110]: Updated. Thanks!

File: tests/management/test_finalize_xblockskill_tags.py (hunk 7 above)
Comment (navinkarkera) [reply tests/management/test_finalize_xblockskill_tags.py:138]: Updated. Thanks!

**Pull request merged**